### PR TITLE
Hurd: Fix bug: pci_get_dev() and dev->aux

### PR DESCRIPTION
**Pull request by jlledom**
Allow clients to read and write from a device w/o a bus scan.
I also fixed some minor indentations problems.

### CODE REVIEW

**Comment by gollux** [comment]
Sorry for the delay, I am ready to merge this soon, but please clean it up as suggested first.

**Comment by gollux** [comment]
I also cleaned up the implementation of pci_fill_info() in all back-ends, including yours. Please see commit 82c06b47dea5a38075ce9d56f743360bc47b4c78 in master for details. Could you please check that it works for you and rebase this branch upon it?

**Comment by jlledom** [comment]
> but please clean it up as suggested first.

What do you mean? Can you refresh my memory? :)

> I also cleaned up the implementation of pci_fill_info() in all back-ends, including yours. Please see commit [82c06b4](https://github.com/pciutils/pciutils/commit/82c06b47dea5a38075ce9d56f743360bc47b4c78) in master for details. Could you please check that it works for you and rebase this branch upon it?

Sure

**Comment by gollux** [comment]
Sorry, Github's review interface is a little bit counter-intuitive and I forgot that I must explicitly publish the review comments.

**Comment by jlledom** [comment]
Hi Martin, found a problem when trying to merge https://github.com/pciutils/pciutils/commit/82c06b47dea5a38075ce9d56f743360bc47b4c78 into my branch. At https://github.com/pciutils/pciutils/blob/master/lib/hurd.c#L310 it's accessing the variable flags which is not in the scope. And the condition to call `hurd_fill_regions()`, at https://github.com/pciutils/pciutils/blob/master/lib/hurd.c#L351, is `flags & (PCI_FILL_BASES | PCI_FILL_SIZES)` which will be true when only one of the flags, or both, are true.

I wonder which was your intention here?

1. Calling `hurd_fill_regions()` only when two flags are true? so the condition is wrong, the flags parameter it's not needed and the check for `PCI_FILL_SIZES` at the end of this function is an oversight.

2. Calling `hurd_fill_regions()` when any of the flags is true. In that case we need to pass flags to the function and check for `PCI_FILL_BASES` as the old `hurd_pci_fill_ino() `did.

I guess it's 1, but please confirm it.

**Comment by gollux** [comment]
Ah, this is indeed wrong. The intention was neither 1 nor 2 :) It was this: Base addresses and sizes are obtained by the same call to `pci_get_dev_regions()`, so whenever we have to call this function (because of `PCI_FILL_BASES` _or_ `PCI_FILL_SIZES`), we can fill both the base and the size. The test of `flags` in `hurd_fill_regions()` shouldn't be there, I will remove it in a moment.

In general, whenever we accidentally obtain any property of the device (e.g., when looking for another property), it always helps to record it in `struct pci_dev` and set the corresponding field flag. This can save future property lookups.

**Comment by gollux** [comment]
I merged your changes and simplified the back-end a little bit more. Could you please check that it still works? :)

**Comment by jlledom** [comment]
Hi, have you pushed your merged changes? I can't see them in the master branch. Anyway, I merged master's changes to my branch (not pushed) and tested it on the Hurd, it works fine.

File: lib/hurd.c [hunk 1]
```diff
@@ -220,7 +233,17 @@ hurd_read(struct pci_dev *d, int pos, byte * buf, int len)
   mach_port_t device_port;
 
   nread = len;
+  if (*((mach_port_t *) d->aux) == 0)
+    {
+      /* We still don't have the port for this device */
+      device_port_lookup(d);
+      if (d->aux == MACH_PORT_NULL)
```
[the remark this answers is not in the document]
Comment (jlledom) [reply lib/hurd.c]: I moved the check inside `device_port_lookup()`, but haven't moved the error messages b/c in `enum_devices()` I need to call `closedir()` before raising the error.

**Comment by gollux** [comment]
Sorry, I pushed it only to my private repo. They should be at Github now.

**Comment by gollux** [comment]
> I moved the check inside `device_port_lookup()`, but haven't moved the error
> messages b/c in `enum_devices()` I need to call `closedir()` before raising
> the error.

Is there any need to call `device_port_lookup()` in `enum_devices()` at all?
It will be done when accessing the device anyway, so I removed it in master.

**Comment by jlledom** [comment]
I wont be able to test this changes in the Hurd until next weekend. I'll tell you if something fails. Thanks!

**Pull request merged**